src/components/NavBar.vue

```diff
@@ -1,31 +1,113 @@
 <template>
-	<nav>
+	<div id="nav-bar">
 		<img src="../assets/logo.svg">
-		<a href="#about">About</a>
-		<a href="#speakers">Speakers</a>
-		<a href="#testimonials">Testimonials</a>
-		<a href="#sponsors">Sponsors</a>
-		<a href="#faq">FAQ</a>
-		<!-- <router-link to='/'>Home</router-link> -->
-		<!-- <router-link :to='{name: "Blog", params: {post: "main"}}'>Read our blog</router-link> -->
-	</nav>
+
+		<label v-if="isMobile" @click="openMenu">&#9776;</label>
+		<nav v-else>
+			<a href="#about">ABOUT</a>
+			<a href="#speakers">SPEAKERS</a>
+			<a href="#testimonials">TESTIMONIALS</a>
+			<a href="#sponsors">SPONSORS</a>
+			<a href="#faq">FAQ</a>
+		</nav>
+
+		<!-- MOBILE MENU -->
+		<div v-if="isMobile && isMenuOpen" class="mobileMenu">
+			<label @click="closeMenu">&#x2715;</label>
+			<nav>
+				<a @click="closeMenu" href="#about">ABOUT</a>
+				<hr noshade>
+				<a @click="closeMenu" href="#speakers">SPEAKERS</a>
+				<hr noshade>
+				<a @click="closeMenu" href="#testimonials">TESTIMONIALS</a>
+				<hr noshade>
+				<a @click="closeMenu" href="#sponsors">SPONSORS</a>
+				<hr noshade>
+				<a @click="closeMenu" href="#faq">FAQ</a>
+			</nav>
+		</div>
+	</div>
 </template>
 
 
 <script>
 export default {
 	name: 'NavBar',
+	data() {
+		return {
+			// ! DOESN'T UPDATE ON SCREEN CHANGE
+			isMobile: screen.width < 850,
+			isMenuOpen: false
+		}
+	},
+	methods: {
+		handleResize() {
+			this.isMobile = window.innerWidth <= 850;
+		},
+		openMenu() {
+			this.isMenuOpen = true; 
+		},
+		closeMenu() {
+			this.isMenuOpen = false;
+		}
+	},
+	beforeCreate() {
+		// ! "THIS" PROBLEMS ARE WEIRD
+		const obj = this;
```

Comment:
instead of setting a variable for `this`, can't you just call this.handleResize()? if you have tried it that way, what kinda of problems do you get when you do it this way?

Reply:
So when you go inside a function scope (as we do in line 57), the definition of `this` changes. It's incredibly weird and one of the few quirks of old JavaScript.

That said, there is a simpler solution: arrow functions! Arrow functions don't change the definition of `this`. So this function can be turned into:

`window.addEventListener('resize', () => this.handleResize());`

And that _should_ work fine.

Reply:
Confirmed, arrow function works. Thanks Naman!